### PR TITLE
Fix: use REST delete instead of patch

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
TIS21-1047

### CODE REVIEW

**Comment by ReubenRobertsHEE** [comment]
Linked PR: https://github.com/Health-Education-England/tis-trainee-details/pull/107

**Review by JosephKelly — APPROVED** [review]
Looks good.
I was wondering if this is getting invoked for everything, e.g. placement deletes and whether that's ok.

**Comment by ReubenRobertsHEE** [comment]
> Looks good.
> I was wondering if this is getting invoked for everything, e.g. placement deletes and whether that's ok.

Good point! We haven't implemented delete for other record types, so I've added a check to limit the API calls. Otherwise we might get a lot of 404 errors (or similar) cluttering up the logs.

File: src/main/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncService.java [hunk 1]
```diff
@@ -54,7 +54,7 @@
   private static final String TABLE_PROGRAMME_MEMBERSHIP = "ProgrammeMembership";
   private static final String TABLE_CURRICULUM = "Curriculum";
 
-  private static final Map<String, String> TABLE_NAME_TO_API_PATH = Map.of(
+  static final Map<String, String> TABLE_NAME_TO_API_PATH = Map.of(
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncService.java:57]: I can't see any use outside of this class.  Do we need to make it package scope?

**Review by JosephKelly — COMMENTED** [review]
Just the 1 question.  Looks good otherwise

File: src/main/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncService.java (hunk 1 above)
Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncService.java]: Whoops, I think I was originally playing around with using it for the test. Rolled that change back.

File: src/test/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncServiceTest.java [hunk 2]
```diff
@@ -414,11 +414,33 @@ void shouldDeleteProgrammeMemberships(Operation operation) {
...
     verify(restTemplate)
-        .patchForObject(anyString(), eq(null), eq(Object.class), eq("programme-membership"),
+        .delete(anyString(), eq("programme-membership"),
             eq("personIdValue"));
     verifyNoMoreInteractions(restTemplate);
   }
 
+  @ParameterizedTest(
+      name = "Should not delete when operation is {0} and table is not "
+          + "ProgrammeMembership")
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncServiceTest.java:424]: Needs to be modified. Currently e.g.:
```
Should not delete when operation is ContactDetails and table is not ProgrammeMembership
```
How about
```
@ParameterizedTest(name = "Should not delete when operation is DELETE and table is {0}")
```

Comment (ReubenRobertsHEE) [reply src/test/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncServiceTest.java]: Good catch. I've also rebased on Ed's work, so there should be no conflicts.

**Review by JosephKelly — APPROVED** [review]

**Pull request merged**